File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int,int>map;
        unordered_set<int>s;

        for(int i:arr){
            map[i]++;
        }
        for(auto &i:map){
            s.insert(map[i.first]);
        }

        if(s.size()==map.size()){
            return true;
        }
        return false;
    }
};

File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: list[int]) -> bool:
        freq={}
        for i in arr:
            freq[i]= freq.get(i,0) +1

        if len(freq.values()) == len(set(freq.values())):
            return True
        return False